cmd/genesis/genesis.go

```diff
@@ -0,0 +1,134 @@
+				ConsensusEngine:               viper.GetString("genesis.consensusEngine"),
+				SmartContractsOwner:           viper.GetString("genesis.smartContractsOwner"),
+				ExtraData:                     viper.GetString("genesis.extraData"),
+			}
+
+			fileName := viper.GetString("genesis.fileName")
+
+			file, err := os.Create(fileName)
+			if err != nil {
+				fmt.Printf("Error generating file: %s", err)
+				os.Exit(1)
+			}
+
+			handler := generateGenesisFileCommandHandler{w: file}
+			err = handler.handle(options)
+			if err != nil {
+				fmt.Printf("Error generating file: %s", err)
+				os.Exit(1)
+			}
+
+			fmt.Println("Genesis file generated.")
+		},
+	}
+
+	cmd.Flags().StringP("config", "c", "", "Use to load configuration from config file.")
+	cmd.Flags().StringP("network", "n", "", "The network to use, test or main")
+	viper.BindPFlag("genesis.network", cmd.Flags().Lookup("network"))
+	cmd.Flags().StringP("maxNumValidators", "v", "", "The maximum num of validators.")
+	viper.BindPFlag("genesis.maxNumValidators", cmd.Flags().Lookup("maxNumValidators"))
+	cmd.Flags().StringP("unbondingPeriod", "p", "", "The unbonding period in days.")
+	viper.BindPFlag("genesis.unbondingPeriod", cmd.Flags().Lookup("unbondingPeriod"))
+	cmd.Flags().StringP("walletAddressGenesisValidator", "g", "", "The wallet address of the genesis validator.")
+	viper.BindPFlag("genesis.walletAddressGenesisValidator", cmd.Flags().Lookup("walletAddressGenesisValidator"))
+	cmd.Flags().StringP("consensusEngine", "e", "", "The consensus engine, right now only supports tendermint")
+	viper.BindPFlag("genesis.consensusEngine", cmd.Flags().Lookup("consensusEngine"))
+	cmd.Flags().StringP("smartContractsOwner", "s", "", "The address of the smart contracts owner.")
+	viper.BindPFlag("genesis.smartContractsOwner", cmd.Flags().Lookup("smartContractsOwner"))
+	cmd.Flags().StringP("extraData", "d", "", "Extra data")
+	viper.BindPFlag("genesis.extraData", cmd.Flags().Lookup("extraData"))
+	cmd.Flags().StringP("prefundedAccounts", "a", "", "The prefunded accounts in format 0x212121:12,0x212121:14")
+	viper.BindPFlag("prefundedAccounts", cmd.Flags().Lookup("prefundedAccounts"))
+	cmd.Flags().StringP("fileName", "o", "genesis.json", "The output filename (default:genesis.json).")
+	viper.BindPFlag("genesis.fileName", cmd.Flags().Lookup("fileName"))
+}
+
+func loadFromFileConfigIfAvailable() {
+	fileConfig, _ := cmd.Flags().GetString("config")
+	if fileConfig != "" {
+		viper.SetConfigFile(fileConfig)
+
+		err := viper.ReadInConfig()
+		if err != nil {
+			panic(fmt.Errorf("Fatal error config file: %s \n", err))
+		}
+	}
+}
+
+func main() {
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func parsePrefundedAccounts(accounts interface{}) []genesis.PrefundedAccount {
+	prefundedAccounts := make([]genesis.PrefundedAccount, 0)
+
+	switch accounts.(type) {
+	// Accounts come from config file.
+	case []interface{}:
+		accountArray := accounts.([]interface{})
```

Comment:
ah, now I get it, maybe the comment wont be necessary if you extract each block of the case to it's own function

`prefundedAccountsFromConfig` and `prefundedAccountsFromCLI`?

```
switch accounts.(type) {
   case []interface{}:
      accountArray := accounts.([]interface{})
      prefundedAccountsFromConfig(accountArray)

   case string:
      accountsString := accounts.(string)
      prefundedAccountsFromCLI(accountsString)
}
```

Reply:
nice